apps/condo/domains/user/schema/ConfirmPhone.js

```diff
@@ -0,0 +1,385 @@
+    adminConfig: {
+        defaultPageSize: 50,
+        maximumPageSize: 200,
+        defaultColumns: 'phone, smsCode, requestedAt, retries',
+    },
+})
+
+
+const ConfirmPhoneActionService = new GQLCustomSchema('ConfirmPhoneActionService', {
+    queries: [
+        { 
+            access: true,
+            schema: 'getPhoneByConfirmPhoneActionToken(token: String!): String',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { token } = args
+                const now = extra.extraNow || Date.now()
+                const { errors, data } = await context.executeGraphQL({
+                    context: context.createContext({ skipAccessControl: true }),
+                    query: `
+                        query findConfirmPhoneFromToken($token: String!, $now: String!) {
+                          confirmPhoneActions: allConfirmPhoneActions(where: { token: $token, expiresAt_gte: $now, completedAt: null }) {
+                            id
+                            phone
+                          }
+                        }
+                    `,
+                    variables: { token, now: new Date(now).toISOString() },
+                })
+                if (errors) {
+                    console.error(errors)
+                    throw new Error('[error]: error while executing find confirm phone action')
+                }
+                if (isEmpty(data.confirmPhoneActions)) {
+                    throw new Error(`${CONFIRM_PHONE_ACTION_EXPIRED}]: Unable to find confirm phone action by token`)
+                }
+                const { phone } = data.confirmPhoneActions[0]
+                return phone
+            },
+        },
+    ],    
+    mutations: [
+        {
+            access: true,
+            schema: 'startConfirmPhoneAction(phone: String!, dv:Int!, sender: JSON!): String',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { phone: inputPhone, sender, dv } = args
+                const phone = normalizePhone(inputPhone)
+                const token = uuid()
+                const now = extra.extraNow || Date.now()
+                const requestedAt = new Date(now).toISOString()
+                const expiresAt = new Date(now + CONFIRM_PHONE_ACTION_EXPIRY * 1000).toISOString()
+                const smsCode = generateSmsCode(phone)
+                const smsCodeRequestedAt = new Date(now).toISOString()
+                const smsCodeExpiresAt = new Date(now + SMS_CODE_TTL * 1000).toISOString()
+                const variables = { 
+                    dv,
+                    sender,
+                    phone,
+                    smsCode,
+                    token,
+                    smsCodeRequestedAt,
+                    smsCodeExpiresAt,
+                    requestedAt,
+                    expiresAt,
+                }
+                const { errors: createErrors } = await context.executeGraphQL({
```

Comment:
it's better to use auto-generated `utils/serverSchema`

Reply:
Done